python_modules/dagster/dagster/_core/definitions/op_selection.py

```diff
@@ -131,23 +141,38 @@ def _get_graph_subset(
                 node.definition,
                 node_selection_tree,
                 parent_handle=node_handle,
+                selected_outputs_by_op_handle=selected_outputs_by_op_handle,
             )
 
         subgraph_nodes[node.name] = node_def
 
+        def is_output_selected(node_output: NodeOutput) -> bool:
+            if not selection_tree.has_child(node_output.node_name):
+                return False
+
+            output_def, op_handle = graph.node_named(
+                node_output.node_name
+            ).definition.resolve_output_to_origin(
+                node_output.output_name, NodeHandle(node_output.node_name, parent_handle)
+            )
+            return (
+                op_handle not in selected_outputs_by_op_handle
```

Comment:
Should this be:
```python
op_handle in selected_outputs_by_op_handle and 
output_def.name in selected_outputs_by_op_handle[op_handle]
```
?

Reply:
The dict only contains entries for ops whose outputs are subsetted. So if an op isn't included in the dict, it means all its outputs are selected.

I'm going to add some docstring to make this more clear:
```python
        selected_outputs_by_op_handle: A mapping of op handles to the set of output names that are
            selected within that op. If an op handle isn't included in this dict, it means that
            _all_ its outputs are selected.
```

Reply:
Ah got it. Makes sense